### DISCUSSION TITLE
Port recent improvements to SVGMobject, CodeMobject etc from 3b1b/manim

### DISCUSSION THREAD

**Discussion by behackl** [ManimCommunity/manim#2817] [Suggestions and Proposals]
## Enhancement proposal

`SVGMobject` and some related mobjects (Text, CodeMobject) have recently been refactored and improved substantially over at 3b1b/manim. It would be a good idea to check in how far these changes could be ported over.

Relevant PRs:

- https://github.com/3b1b/manim/pull/1731
- https://github.com/3b1b/manim/pull/1745
- https://github.com/3b1b/manim/pull/1751

**Comment by naveen521kk** [comment]
I would say not to port over https://github.com/3b1b/manim/pull/1751 (I would be working on the new text API soon), others yeah it's good to have.

**Comment by YishiMichael** [comment]
Recently I've finished refactoring `MTex`, `MarkupText`, both of which inherit from `StringMobject`, a newly-constructed ABC. I'd like to hear some suggestions from Manim Community, and seek for possibility to port it over here.

See PR here:
- https://github.com/3b1b/manim/pull/1795

**Comment by naveen521kk** [comment]
For any changes in `Text`/`MarkupText` waiting for https://github.com/ManimCommunity/manim/issues/2355 is a great idea. It would be great if you could help out with how it would work with the API proposed there. 

For the `MTex`/`Tex` you can open a PR and we'll discuss it there. Thanks for your contribution!
